cdap-proto/src/main/java/co/cask/cdap/proto/ProtoTrigger.java

```diff
@@ -142,17 +142,19 @@ public String toString() {
 
   /**
    * Abstract base class for composite trigger in REST requests/responses.
+   *
+   * @param <T> type of triggers contained in the composite trigger
    */
-  public abstract static class AbstractCompositeTrigger extends ProtoTrigger {
-    private final List<Trigger> triggers;
+  public abstract static class AbstractCompositeProtoTrigger<T extends Trigger> extends ProtoTrigger {
+    private final List<T> triggers;
 
-    public AbstractCompositeTrigger(Type type, Trigger... triggers) {
+    public AbstractCompositeProtoTrigger(Type type, List<T> triggers) {
```

Comment:
I think in a previous comment I asked you to rename this, but I've since learned that we have backwards compatibility requirement for cdap-proto. Sorry, but please rename these back to what they were before, but keep the new generic.

Reply:
I can rename the internal composite trigger instead. How about calling the internal one `AbstractSatisfiableCompositeTrigger`?

Reply:
makes sense